packages/components/src/components/molecules/CardPile/CardPile.tsx

```diff
@@ -1,122 +1,152 @@
-        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
-        <TagName {...cardProps} ref={firstCardRef} />
-      </div>
-    );
-  }, [thisCard, classes, component]);
-
-  const backFaceComponent = useMemo(() => {
-    const BackFaceTagName = backFace;
-    return pileCards.length <= 1 ? undefined : (
-      <div data-testid="CardPile-backface" className={classnames(classes.card, classes.lastItem)}>
-        <BackFaceTagName />
-      </div>
-    );
-  }, [pileCards.length, classes, backFace]);
+  const [isSelected, setSelect] = useState(false);
+  const [firstCard, secondCard, ...otherCards] = cards;
 
   const toggleSelect = () => {
     setSelect(!isSelected);
   };
-  const onPileClick = () => toggleSelect();
-  const onPileKeyDown = (event: KeyboardEvent<HTMLDivElement>) => event.keyCode === 32 && toggleSelect();
-  const onTurnFirstCard = () => firstCardRef.current?.flip();
-  const onShuffle = () => {
-    firstCardRef.current?.flip(false);
-    const newPile = shuffleArray(pileCards);
-    setPileCards(newPile);
+
+  const onTurnFirstCard = () => {
+    onCardFlipped(!isFaceUp);
+    setSelect(false);
+  };
+
+  const onShufflePile = () => {
+    onShuffle();
+    setSelect(false);
   };
 
   return (
-    <div
+    <Draggable
+      className={classes.cardPile}
+      position={position}
+      height={height}
+      width={width}
+      boundaries={boundaries}
+      onDragged={onPositionChanged}
+      onClick={toggleSelect}
+      // eslint-disable-next-line react/destructuring-assignment
       data-testid="CardPile"
-      role="button"
-      tabIndex={0}
-      onClick={onPileClick}
-      onKeyDown={onPileKeyDown}
-      className={classnames(classes.cardPile, { [classes.multiItem]: pileCards.length > 1 })}
     >
       {isSelected && (
         <>
           <Overlay />
           <div className={classes.cardPileMenu}>
-            <CardPileMenu onTurnFirstCard={onTurnFirstCard} onShufflePile={onShuffle} />
+            <CardPileMenu onTurnFirstCard={onTurnFirstCard} onShufflePile={onShufflePile} />
           </div>
         </>
       )}
-      {displayedCard}
-      {backFaceComponent}
-    </div>
+      {firstCard && (
+        <BaseCard
+          data-testid="CardPile-firstCard"
+          className={classes.firstItem}
+          width={width}
+          height={height}
+          frontFace={<PlayingCardFrontFace card={firstCard as PlayingCard} />}
+          backFace={<PlayingCardBackFace />}
+          disableNativeEvents
+          faceUp={isFaceUp}
+        />
+      )}
+      {secondCard && (
+        <BaseCard
+          data-testid="CardPile-secondCard"
+          className={classes.secondItem}
+          width={width}
+          height={height}
+          frontFace={<PlayingCardFrontFace card={secondCard as PlayingCard} />}
+          backFace={<PlayingCardBackFace />}
+          disableNativeEvents
+          faceUp={false}
+        />
+      )}
+      {otherCards.map((card, index) => {
```

Comment:
I guess it's a bit overkill to create a div for each card no?
I mean, in terms of performance, create 3 divs or 72 have a small difference - also for dragging it

Reply:
Honestly I couldn't find a way to generate the style of "depth" of the card pile depending on how many cards it has.

![image](https://user-images.githubusercontent.com/10576609/100539470-fb429680-3236-11eb-87ce-4671014c4c02.png)

Also I checked and it does not affect performance when dragging it, maybe in rendering it would affect if we had like 10000 decks but it will probably never be the case 🤷‍♂️

Reply:
Yeah indeed, the DOM wouldn't be a problem on our context